app/code/Magento/Bundle/Model/Product/Type.php

```diff
@@ -536,15 +536,17 @@ public function updateQtyOption($options, \Magento\Framework\DataObject $option,
 
         foreach ($selections as $selection) {
             if ($selection->getProductId() == $optionProduct->getId()) {
-                foreach ($options as &$option) {
-                    if ($option->getCode() == 'selection_qty_' . $selection->getSelectionId()) {
+                foreach ($options as &$quoteItemOption) {
+                    if ($quoteItemOption->getCode() == 'selection_qty_' . $selection->getSelectionId()) {
                         if ($optionUpdateFlag) {
-                            $option->setValue(intval($option->getValue()));
+                            $quoteItemOption->setValue(intval($option->getValue()));
                         } else {
-                            $option->setValue($value);
+                            $quoteItemOption->setValue($value);
                         }
                     }
                 }
+
+                unset($quoteItemOption);
```

Comment:
https://secure.phabricator.com/book/phabflavor/article/php_pitfalls/

The easiest way to avoid this is to avoid using foreach-by-reference. If you do use it, unset the reference after the loop:

foreach ($array as &$value) {
  // ...
}
unset($value);